certinel.go

```diff
@@ -65,15 +64,6 @@ func (c *Certinel) Close() error {
 // can be passed as the GetCertificate member in a tls.Config object. It is
 // safe to call across multiple goroutines.
 func (c *Certinel) GetCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	return c.certificate, nil
-}
-
-func (c *Certinel) loadCertificate(certificate tls.Certificate) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	c.certificate = &certificate
+	cert, _ := c.certificate.Load().(*tls.Certificate)
```

Comment:
~~Are we ignoring an error here?~~ nope, but we are not going to panic either.

If this fails were going to return (nil, nil) from the function which isn't something that normally happens. I think in this case we need to check the ok return and properly return an error.